### PR TITLE
IGNITE-13197: Import order rule added.

### PR DESCRIPTION
**Pull request by nizhikov**
PR  contains new checkstyle rule to check import order according to Ignite Code Style.
[IntelliJ IDEA setup to match the check](https://yadi.sk/i/kyXSE5rDc3ePOA)

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

**Comment by zaleslaw** [comment]
Could we create an XML checkstyle for IDEA or something else to make automatic refactoring?

**Comment by nizhikov** [comment]
@zaleslaw If you setup IDEA equal to [screenshot](https://yadi.sk/i/kyXSE5rDc3ePOA) then it will automatically rearrange imports according to codestyle(Code -> Optimize imports). All new imports will goes to the right place, also.

**Comment by zaleslaw** [comment]
So, I agree, but for newcomers it's a good point to have automatic
checkstyle with all possible rules. I had one a year ago, but it doesn't
cover all cases.
If you have one, please let me know

вт, 30 июн. 2020 г. в 17:51, Nikolay <notifications@github.com>:

**Comment by nizhikov** [comment]
@zaleslaw We have automation in the form of maven `checkstyle` profile. The goal of this PR is to extend the coverage of automatic checks with the automatic Import Order rule. AFAIK checkstyle+maven is common solution and used in many other open-source projects.

Do you have in mind some other kind of automation?

**Comment by zaleslaw** [comment]
No automation, maybe it sounds wrong earlier: I suggest only shared
community checkstyle.xml for using in IDEA (somebody shared one with me a
few year ago, but now I lost it after moving on another laptop)

вт, 30 июн. 2020 г. в 18:03, Nikolay <notifications@github.com>:

**Comment by nizhikov** [comment]
We have this file and it updated in PR. 
Changes in settings.xml is the essence of PR

**Comment by zaleslaw** [comment]
Great, look like I missed the xml and saw only reordered  imports  in files

вт, 30 июн. 2020 г. в 18:16, Nikolay <notifications@github.com>:

**Comment by pavlukhin** [comment]
@nizhikov I see that travis build check failed and it seems that it is related to the import order rule (https://travis-ci.org/github/apache/ignite/builds/703548289?utm_source=github_status&utm_medium=notification). Could you please take a look?

File: modules/direct-io/src/main/java/org/apache/ignite/internal/processors/cache/persistence/file/IgniteNativeIoLib.java [hunk 1]
```diff
@@ -150,10 +150,10 @@ else if (verIntComps.get(verIdx) == 2) {
 
             if (compsCnt > majorRevIdx && verIntComps.get(majorRevIdx) > 4)
                 return true;
-            else if (compsCnt > minorRevIdx
-                && verIntComps.get(majorRevIdx) == 4
-                && verIntComps.get(minorRevIdx) >= 10)
-                return true;
+            else
```
Comment (ololo3000) [inline modules/direct-io/src/main/java/org/apache/ignite/internal/processors/cache/persistence/file/IgniteNativeIoLib.java:153]: It seems that braces are required here according to https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines#CodingGuidelines-BracesandIdentation

Comment (nizhikov) [reply modules/direct-io/src/main/java/org/apache/ignite/internal/processors/cache/persistence/file/IgniteNativeIoLib.java]: Good cactch! Actually, this changes are not related to the PR.
So it has to be reverted. Thanks!

**Pull request merged**